confidence/confidence_internal_test.go

```diff
@@ -0,0 +1,260 @@
+package confidence
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"reflect"
+	"fmt"
+	"testing"
+	"github.com/stretchr/testify/assert"
+)
+
+type MockResolveClient struct {
+	MockedResponse ResolveResponse
+	MockedError    error
+	TestingT       *testing.T
+}
+
+func (r MockResolveClient) SendResolveRequest(_ context.Context,
+	request ResolveRequest) (ResolveResponse, error) {
+	assert.Equal(r.TestingT, "user1", request.EvaluationContext["targeting_key"])
+	return r.MockedResponse, r.MockedError
+}
+
```

Comment:
Would be super happy if we had some tests for the parent -> child context inheritance and shadowing stuff. 🙏

Reply:
will add some.

Reply:
done.